src/app/pages/services/components/service-s3/service-s3.component.ts

```diff
@@ -1,205 +1,132 @@
-          this.certificate.options.push({ label: item.name, value: item.id });
-        });
-      }
+  ngOnInit(): void {
+    this.ws.call('s3.config').pipe(untilDestroyed(this)).subscribe((config) => {
+      this.form.patchValue(config, { emitEvent: false });
+      this.initialPath = config.storage_path;
     });
-    this.ws
-      .call('s3.bindip_choices')
-      .pipe(
-        map((response) => {
-          return Object.keys(response || {}).map((key) => ({
-            label: response[key],
-            value: key,
-          }));
-        }),
-      )
-      .pipe(untilDestroyed(this)).subscribe((choices) => {
-        choices.forEach((ip) => {
-          this.validBindIps.push(ip.value);
-        });
 
-        const config = _.find(this.fieldConfig, { name: 'bindip' }) as FormSelectConfig;
-        config.options = choices;
-      });
-    entityForm.submitFunction = this.submitFunction;
-  }
+    this.form.controls['storage_path'].valueChanges.pipe(untilDestroyed(this)).subscribe((newPath) => {
+      if (!newPath || newPath === this.initialPath || this.warned) {
+        return;
+      }
 
-  resourceTransformIncomingRestData(data: any): any {
-    if (data.certificate && data.certificate.id) {
-      data['certificate'] = data.certificate.id;
-    }
-    if (data.storage_path) {
-      this.initial_path = data.storage_path;
-    }
+      this.dialog
+        .confirm({
+          title: helptext.path_warning_title,
+          message: helptext.path_warning_msg,
+        })
+        .pipe(untilDestroyed(this))
+        .subscribe((confirmed) => {
+          if (!confirmed) {
+            this.form.patchValue({ storage_path: this.initialPath });
+          }
 
-    // If validIps is slow to load, skip check on load (It's still done on save)
-    if (this.validBindIps.length > 0) {
-      return this.compareBindIps(data);
-    }
-    return data;
+          this.warned = true;
+        });
+    });
   }
 
-  compareBindIps(data: any): any {
-    if (data.bindip && this.validBindIps.length > 0) {
-      if (!this.validBindIps.includes(data.bindip)) {
-        data.bindip = '';
-      }
-    }
-    return data;
-  }
+  onSubmit(): void {
+    const values = {
+      ...this.form.value,
+      bindport: Number(this.form.value.bindport),
+    };
 
-  submitFunction(configUpdate: S3ConfigUpdate): Observable<S3Config> {
-    return this.ws.call('s3.update', [configUpdate]);
+    this.isFormLoading = true;
+    this.ws.call('s3.update', [values])
+      .pipe(untilDestroyed(this))
+      .subscribe(() => {
+        this.isFormLoading = false;
+        this.router.navigate(['/services']);
+        this.cdr.markForCheck();
+      }, (error) => {
+        this.isFormLoading = false;
+        this.errorHandler.handleWsFormError(error, this.form);
+        this.cdr.markForCheck();
+      });
   }
 
-  beforeSubmit(data: any): void {
-    this.compareBindIps(data);
+  onCancel(): void {
+    this.router.navigate(['/storage']);
```

Comment:
Shouldn't it be `/services`? Other forms usually redirect back to the Services page

Reply:
It should